### PR TITLE
Validate name size

### PR DESCRIPTION
**Pull request by embhorn**
This issue was reported in ZD11703

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -45557,6 +45557,9 @@ int wolfSSL_X509_NAME_print_ex(WOLFSSL_BIO* bio, WOLFSSL_X509_NAME* name,
             return WOLFSSL_FAILURE;
     }
 
+    if (name->sz == 0)
```
Comment (SparkiDev) [inline src/ssl.c:45560]: Further down sz is used with -1 and -2.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (embhorn) [reply src/ssl.c:45560]: Added more checks in appropriate conditions

Comment (dgarske) [reply src/ssl.c:45560]: This API `wolfSSL_X509_NAME_print_ex` is public and we are not checking `name` for NULL... Please add.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (embhorn) [reply src/ssl.c:45560]: Check

**Review by dgarske — APPROVED** [review]

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**